SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift

```diff
@@ -0,0 +1,190 @@
+        }
+    }
+}*/
+func startConnectionSpan(request: URLRequest?) -> Span? {
+    if request == nil || request?.url == nil {
+        return nil
+    }
+    let url = request!.url!
+    if !(url.scheme?.lowercased().starts(with: "http") ?? false) {
+        return nil
+    }
+    let method = request!.httpMethod ?? "GET"
+    // Don't loop reporting on communication with the beacon
+    let absUrlString = url.absoluteString
+    if SplunkRum.theBeaconUrl != nil && absUrlString.starts(with: SplunkRum.theBeaconUrl!) {
+        return nil
+    }
+    if SplunkRum.configuredOptions?.ignoreURLs != nil {
+        let result = SplunkRum.configuredOptions?.ignoreURLs?
+            .matches(in: absUrlString, range: NSRange(location: 0,
+                                                      length: absUrlString.utf16.count))
+        if result?.count != 0 {
+            return nil
+        }
+    }
+    let tracer = buildTracer()
+    let span = tracer.spanBuilder(spanName: "HTTP "+method).setSpanKind(spanKind: .client).startSpan()
+    span.setAttribute(key: "component", value: "http")  // "NSURLConnection"
+    span.setAttribute(key: "http.url", value: url.absoluteString)
+    span.setAttribute(key: "http.method", value: method)
+    if let body = request?.httpBody {
+        span.setAttribute(key: "http.request_content_length", value: Int(body.count))
+    }
+    return span
+}
+
+func endConnectionSpan(connection: NSURLConnection?,
+                       status: String,
+                       hresponse: HTTPURLResponse?,
+                       error: Error?,
+                       span: Span = OpenTelemetry.instance.contextProvider.activeSpan!) {
+    if hresponse != nil {
+        span.setAttribute(key: "http.status_code", value: hresponse!.statusCode)
+        for (key, val) in hresponse!.allHeaderFields {
+            let keyStr = key as? String
+            if keyStr != nil {
+                if keyStr?.caseInsensitiveCompare("server-timing") == .orderedSame {
+                    let valStr = val as? String
+                    if valStr != nil {
+                        if valStr!.starts(with: "traceparent") {
+                            addLinkToSpan(span: span, valStr: valStr!)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if error != nil {
+        span.setAttribute(key: "error", value: true)
+        span.setAttribute(key: "exception.message", value: error!.localizedDescription)
+        span.setAttribute(key: "exception.type", value: String(describing: type(of: error!)))
+    }
+    span.setAttribute(key: "http.response_content_length_uncompressed",
+                      value: Int(hresponse?.expectedContentLength ?? 0))
+    if hostConnectionType != nil {
+        span.setAttribute(key: "net.host.connection.type", value: hostConnectionType!)
+    }
+    span.end()
+}
+func swizzleClassMethod(clazz: AnyClass, orig: Selector, swizzled: Selector) {
```

Comment:
Think we can move these helpers to a shared file since `NetworkInstrumentation` is using the exact same code

Reply:
This method is different than  NetworkInstrumentation method,  swizzleClassMethod method is used to swizzle class method and NetworkInstrumentation method is for swizzle instance method.